src/m365/commands/login.ts

```diff
@@ -238,6 +246,38 @@ class LoginCommand extends Command {
     this.initAction(args, logger);
     await this.commandAction(logger, args);
   }
+
+  private shouldRenewConnection(options: Options): boolean {
+    const authType = options.authType || cli.getSettingWithDefaultValue<string>(settingsNames.authType, 'deviceCode');
+    const ensure: boolean | undefined = options.ensure;
+
+    if (!ensure || (ensure && (
+      (!auth.connection.active && AuthType[authType as keyof typeof AuthType] !== auth.connection.authType) ||
+      (options.userName && options.userName !== auth.connection.userName) ||
+      (options.certificateFile && (auth.connection.certificate !== fs.readFileSync(options.certificateFile as string, 'base64'))) ||
+      (options.appId && options.appId !== auth.connection.appId) ||
+      (options.tenant && options.tenant !== auth.connection.tenant)))) {
+      return true;
+    }
+
+    return false;
+  }
+
+  private async obtainAccessToken(logger: Logger): Promise<void> {
```

Comment:
Let's rename this method to `getAccessToken`. _get_ is an equivalent of _obtain_ that's easier to ready for non-fluent English speakers.